### PR TITLE
openstack: Test Nova AZ manifest generation

### PR DESCRIPTION
**Pull request by pierreprinetti**
* instrument the manifest-generation test-framework to execute Python
testfiles
* test machinesets with availability zones
only)

### CODE REVIEW

File: hack/openstack/test-manifests.sh [hunk 1]
```diff
@@ -12,6 +12,10 @@ declare \
 	external_network='' \
 	compute_flavor=''
 
+# Let install-config describe a configuration that is incompatible with the
+# target CI infrastructure
+export OPENSHFIT_INSTALL_SKIP_PREFLIGHT_VALIDATIONS=1
```
Comment (mandre) [inline hack/openstack/test-manifests.sh:17]: Perhaps we should remove this export from https://github.com/openshift/release/pull/13950 if you're adding it to the `test-manifests.sh` script directly?

File: scripts/openstack/manifest-tests/nova-availability-zones/test_machinesets.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+import sys
+import glob
+import yaml
+
+ASSETS_DIR = ""
+
+EXPECTED_MACHINESET_NUMBER = 3
+EXPECTED_MACHINES_NUMBER = 1000
+EXPECTED_ZONE_NAMES = ["zone", "ztwo", "zthree"]
```
Comment (mandre) [inline scripts/openstack/manifest-tests/nova-availability-zones/test_machinesets.py:14]: :rofl:

File: scripts/openstack/manifest-tests/nova-availability-zones/test_machinesets.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+import unittest
+
+import sys
+import glob
+import yaml
+
+ASSETS_DIR = ""
+
+EXPECTED_MACHINESET_NUMBER = 3
```
Comment (mandre) [inline scripts/openstack/manifest-tests/nova-availability-zones/test_machinesets.py:12]: This constant isn't used.

File: hack/openstack/test-manifests.sh (hunk 1 above)
Comment (pierreprinetti) [reply hack/openstack/test-manifests.sh:17]: done :)

**Comment by pierreprinetti** [comment]
@mandre Addressed your comments! PTAL

**Review by mandre — COMMENTED** [review]
/lgtm

**Comment by staebler** [comment]
I don't think that [hack/openshift/OWNERS](https://github.com/openshift/installer/blob/master/hack/openstack/OWNERS) file is working as a symbolic link.

/approve

**Pull request merged**